server/channels/api4/image.go

```diff
@@ -37,10 +37,17 @@ func getImage(c *Context, w http.ResponseWriter, r *http.Request) {
 		parsedURL.Host = siteURL.Host
 	}
 
-	// in case image proxy is enabled and we are fetching a remote image (NOT static or served by plugins), pass request to proxy
-	if *c.App.Config().ImageProxySettings.Enable && parsedURL.Host != siteURL.Host {
-		c.App.ImageProxy().GetImage(w, r, parsedURL.String())
-	} else {
-		http.Redirect(w, r, parsedURL.String(), http.StatusFound)
+	if *c.App.Config().ImageProxySettings.Enable {
+		// in case image proxy is enabled and we are fetching a remote image (NOT static or served by plugins), pass request to proxy
+		if parsedURL.Host != siteURL.Host {
+			c.App.ImageProxy().GetImage(w, r, parsedURL.String())
+		} else {
+			// Otherwise we redirect.
+			http.Redirect(w, r, parsedURL.String(), http.StatusFound)
+		}
+		return
 	}
+
+	// When proxy disabled this endpoint should fail as we don't support redirecting to external images any longer (MM-54477).
+	c.Err = model.NewAppError("getImage", "api.image.get.app_error", nil, "", http.StatusBadRequest)
```

Comment:
Is there a different status code that we should be using for this? We used to use `http.StatusNotImplemented` everywhere that an API was disabled by a config setting, and we still do in plenty of places, but I remember having a discussion about how 500 errors affected SLAs so we didn't want to use them for things like that.

Reply:
Also, nitpicking, but I'd normally expect the error case to come before the normal codepath like
```go
if !*c.App.Config().ImageProxySettings.Enable {
    c.Err = model.NewAppError()
    return
}

if parsedURL.Host != siteURL.Host 
    // ...
} else {
    // ...
}
```

Reply:
@hmhealey I think a 4xx code is appropriate here since the client should avoid making requests to this endpoint with external URLs. The parameter in the request (URL) defines the success of it. In other words, I don't see this as a server's deficiency since there's no additional functionality to implement here.

> Also, nitpicking, but I'd normally expect the error case to come before the normal codepath

Fair. This is a thing I tend to do when security is involved, which is making the secure (happy) path the exception rather than the default case. Overall I think that with proper comments in place both approaches should be equally acceptable.

Reply:
I don't think the client should actually be using this for internal images anyway. I forget the exact reason that this can handle internal images, but it feels like it was an edge case.

Regardless of that though, 400 seems totally fine since the end result is still that the this doesn't. It shouldn't be used when the proxy is disabled either way, so this is just plugging the hole of someone misusing it